snappy/kernel_os.go

```diff
@@ -68,6 +68,15 @@ func extractKernelAssets(s *snap.Info, snapf snap.File, flags InstallFlags, inte
 		return fmt.Errorf("can not extract kernel assets from snap type %q", s.Type)
 	}
 
+	// sanity check that we have the new kernel format
+	kmeta, err := snapf.ReadFile("meta/kernel.yaml")
```

Comment:
as we discussed snapf is probably not needed here (we already mounted the snap at this point) which would simplify things, also may be that we don't Unpack in snap.File (may still be good to have around until we really know more)